python/cudf/cudf/tests/test_index.py

```diff
@@ -323,3 +323,10 @@ def test_index_names():
 )
 def test_range_index_from_range(data):
     assert_eq(pd.Index(data), cudf.core.index.as_index(data))
+
+
+def test_empty_df_head_tail_index():
```

Comment:
Can you add a test for slicing 10 elements off of a 5 element dataframe as well?